src/components/TodoList/TodoList.tsx

```diff
@@ -0,0 +1,150 @@
+import React, { useState } from 'react';
+import cn from 'classnames';
+import { Todo } from '../../types/Todo';
+
+interface Props {
+  todos: Todo[],
+  handleDeleteTodo: (todo: number) => void;
+  handleComplitedTodo: (todoId: number, toggle: boolean) => void;
+  editingTodosId: number[];
+  formShowedForId: number;
+  showForm: (todoId: number) => void;
+  handleUpdateTodoTitle: (todoId: number, newTitle: string) => void,
+  tempTodo: Todo | null,
+}
+
+export const TodoList: React.FC<Props> = ({
+  todos,
+  handleDeleteTodo,
+  handleComplitedTodo,
+  editingTodosId,
+  formShowedForId,
+  showForm,
+  handleUpdateTodoTitle,
+  tempTodo,
+}) => {
+  const [newTitle, setNewTitle] = useState('');
+
+  const handleTitleChange = (todo: Todo) => {
+    if (newTitle === todo.title) {
+      showForm(0);
+      setNewTitle('');
+    } else if (!newTitle.trim()) {
+      handleDeleteTodo(todo.id);
+    } else {
+      handleUpdateTodoTitle(todo.id, newTitle);
+      showForm(0);
+    }
+  };
+
+  const cancelTitleCahnge = (event: React.KeyboardEvent) => {
+    if (event.key === 'Escape') {
+      showForm(0);
+      setNewTitle('');
+    }
+  };
+
+  const tempTodoItem = (
+    <div
+      className="todo"
+    >
+      <label className="todo__status-label">
+
+        <input
+          type="checkbox"
+          className="todo__status"
+        />
+
+      </label>
+
+      <span className="todo__title">
+        {tempTodo?.title}
+      </span>
+
+      <div className="modal overlay is-active">
+        <div className="modal-background has-background-white-ter" />
+        <div className="loader" />
+      </div>
+    </div>
+  );
+
+  return (
+    <section className="todoapp__main">
+
+      {todos.map(todo => {
+        const formTitleChange = formShowedForId === todo.id
```

Comment:
so many different component in one `map`
will be better to create separate component for each TodoItem